tedana/decomposition/gift_pca.py

```diff
@@ -0,0 +1,599 @@
+"""
+PCA based on GIFT software
+"""
+import logging
+
+import numpy as np
+from sklearn.decomposition import PCA, TruncatedSVD
+from sklearn.preprocessing import StandardScaler
+
+from scipy.signal import detrend
+from scipy.fftpack import fft, fftshift, fftn, fft2
+from scipy.signal import correlate2d
+
+LGR = logging.getLogger(__name__)
+
+
+def _autocorr(x):
+    """
+    Run Singular Value Decomposition (SVD) on input data,
+    automatically select components based on the GIFT software.
+
+    Parameters
+    ----------
+    x :
+
+    Returns
+    -------
+    u :
+    """
+    result = np.correlate(x, x, mode='full')
+    return result[result.size / 2:]
+
+
+def _sumN(dat):
+    """
+    Sum of all the elements of the dat matrix.
+
+    Parameters
+    ----------
+    dat :
+
+    Returns
+    -------
+    u :
+    """
+    return np.sum(dat[:])
+
+
+def _checkOrder(n_in):
+    """
+    Checks the order passed to the window functions.
+
+    Parameters
+    ----------
+    n_in :
+
+    Returns
+    -------
+    n_out :
+    w :
+    trivalwin :
+    """
+
+    w = []
+    trivalwin = 0
+
+    # Special case of negative orders:
+    if n_in < 0:
+        raise ValueError('Order cannot be less than zero.')
+
+    # Check if order is already an integer or empty
+    # If not, round to nearest integer.
+    if not n_in or n_in == np.floor(n_in):
+        n_out = n_in
+    else:
+        n_out = np.round(n_in)
+        LGR.warning('Rounding order to nearest integer.')
+
+    # Special cases:
+    if not n_out or n_out == 0:
+        w = np.zeros((0, 1))  # Empty matrix: 0-by-1
+        trivalwin = 1
+    elif n_out == 1:
```

Comment:
Stylistic nit-pick: can we change this to an `else` instead of an `elif` ?

Reply:
Same thing I said before. `n_out` can be higher than one, which is why `trivalwin = 0`is possible.